### PR TITLE
271 implement restore password

### PR DESCRIPTION
**Pull request by MatyashDen**

### CODE REVIEW

File: common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt [hunk 1]
```diff
@@ -61,4 +61,16 @@ class AuthRequests {
         object Success : Action
         data class Failure(override val message: Message) : ToastAction
     }
+
+    class SendPasswordReset(private val email: String) : Request() {
+
+        override suspend fun execute() = firebaseController.sendPasswordReset(email) { e ->
```
Comment (bogdan-evtushenko) [inline common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt:67]: What is `e`?

File: common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt [hunk 2]
```diff
@@ -61,4 +61,16 @@ class AuthRequests {
...
+            e?.let {
+                store.dispatch(Failure(e.message.toMessage()))
+            } ?: store.dispatch(Success("Email with further instructions was sent to you! Please check!".toMessage()))
```
Comment (bogdan-evtushenko) [inline common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt:70]: `has been sent`?

**Review by bogdan-evtushenko — APPROVED** [review]

File: common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt (hunk 1 above)
Comment (MatyashDen) [reply common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt:67]: exception

Comment (bogdan-evtushenko) [reply common/src/commonMain/kotlin/io/xorum/codeforceswatcher/features/auth/redux/AuthRequests.kt:67]: not clear

**Pull request merged**